### PR TITLE
Initial commit legacy materials

### PR DESCRIPTION
**Pull request by zicher3d**
This commit includes:
 - all Legacy (Protein) classes
 - procedural nodes (checker, noise, turbulence 2d/3d)
 - utilities (stain, normals combine nodes)

For now all nodes are under "adsk_legacy" group, except the turbulence ones that might be ok to be added to the standard lib.

Because of a bug in the handling of namespaces in the Graph Editor all namespaces are set to "".
Bug: https://github.com/AcademySoftwareFoundation/MaterialX/issues/1695

### CODE REVIEW

**Comment by ashwinbhat** [comment]
@zicher3d can you include an example mtlx for legacy

**Comment by zicher3d** [comment]
Will do. I have one that works in MaterialX viewer. I need to update it.
Then I just need to do new commits to the branch and they will add to the PR?

**Review by nicolassavva-autodesk — COMMENTED** [review]
added a few quick inline comments

File: contrib/adsk/libraries/adsklib/genglsl/mx_turbulence2d_float.glsl [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include "../../stdlib/genglsl/lib/mx_noise.glsl"
```
[the remark this answers is not in the document]
Comment (zicher3d) [reply contrib/adsk/libraries/adsklib/genglsl/mx_turbulence2d_float.glsl:1]: The file is in the adsklib folder, not stdlib, so I need that path.

File: contrib/adsk/libraries/adsklib/genglsl/adsklib_genglsl_impl.mtlx [hunk 2]
```diff
@@ -7,4 +7,8 @@
   -->
   <!-- <backface_util> -->
   <implementation name="IM_backface_util_genglsl" nodedef="adsk:ND_backface_util" file="mx_backface_util.glsl" function="mx_backface_util" target="genglsl"/>
+  
+  <implementation name="IM_turbulence2d_float_genglsl" nodedef="ND_turbulence2d_float" file="mx_turbulence2d_float.glsl" function="mx_turbulence2d_float" target="genglsl" />
+  <implementation name="IM_turbulence3d_float_genglsl" nodedef="ND_turbulence3d_float" file="mx_turbulence3d_float.glsl" function="mx_turbulence3d_float" target="genglsl" />
```
[the remark this answers is not in the document]
Comment (zicher3d) [reply contrib/adsk/libraries/adsklib/genglsl/adsklib_genglsl_impl.mtlx:12]: Yes, the current stdlib noise types are Perlin, Cell, Worley and Fractal

File: contrib/adsk/libraries/adsklib/adsklib_legacy_ng.mtlx [hunk 3]
```diff
@@ -0,0 +1,2219 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<materialx version="1.38">
+  <!--
+  DESCRIPTION: Nodegraphs for Autodesk Legacy Material Classes
+  VERSION: 0.9.0
```
[the remark this answers is not in the document]
Comment (zicher3d) [reply contrib/adsk/libraries/adsklib/adsklib_legacy_ng.mtlx:5]: Definetly. We can discuss that. Also I think we need an SHA value. Not sure about that process yet.

**Review by ashwinbhat — APPROVED** [review]
LTGM We might have to revisit the nodegroup and namespace at a later point.

**Comment by nicolassavva-autodesk** [comment]
Let's squash the commits from this PR and merge.
The additions are isolated to the mtlx library and implementations so the prototype should be safe to try out
even if some finishing touches remain before this becomes production ready
(please version the newly introduced nodes accordingly)

**Review by nicolassavva-autodesk — APPROVED** [review]
Go ahead and squash-merge

**Pull request merged**